plugins/amongus/Game.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "Game.h"
+
+#include "mumble_plugin_utils.h"
+
+Game::Game(const procid_t id, const std::string name) : m_ok(false), m_proc(id, name) {
+	if (!m_proc.isOk()) {
+		return;
+	}
+
+	const auto &modules = m_proc.modules();
+	const auto iter     = modules.find("GameAssembly.dll");
+	if (iter == modules.cend()) {
+		return;
+	}
+
+	// 74 89             jz     short loc_????????
+	// A1 ?? ?? ?? ??    mov    eax, AmongUsClient_c **
+	// 8B 40 5C          mov    eax, [eax+5Ch]
+	const std::vector< uint8_t > clientPattern = { 0x74, 0x39, 0xA1, '?', '?', '?', '?', 0x8B, 0x40, 0x5C };
+	m_client                                   = m_proc.findPattern(clientPattern, iter->second);
+	if (!m_client) {
+		return;
+	}
+
+	m_client = m_proc.peekPtr(m_proc.peekPtr(m_client + 3));
```

Comment:
Is `+ 3` the offset to reach the `?`s inside `clientPattern`?

Reply:
Yep!

Reply:
Then maybe this could be mentioned in a comment or something like that?

Reply:
Isn't the comment above with the assembly code enough?

Reply:
It provides enough information to understand what's going on if one is to start starring at the code. If someone simply skims over it, I think adding a comment like `// +3 in order to skip to the memory address we actually care about` would make it much easier to understand what is happening :shrug: